packages/ckeditor5-engine/src/model/utils/findoptimalinsertionrange.ts

```diff
@@ -14,29 +14,29 @@ import type Model from '../model';
-//
-// For instance, if a selection is in the middle of a paragraph, the collapsed range before this paragraph
-// will be returned so that it is not split. If the selection is at the end of a paragraph,
-// the collapsed range after this paragraph will be returned.
-//
-// Note: If the selection is placed in an empty block, the range in that block will be returned. If that range
-// is then passed to {@link module:engine/model/model~Model#insertContent}, the block will be fully replaced
-// by the inserted widget block.
-//
-// **Note:** Use {@link module:widget/utils#findOptimalInsertionRange} instead of this function outside engine.
-// This function is only exposed to be used by {@link module:widget/utils#findOptimalInsertionRange findOptimalInsertionRange()}
-// in the `widget` package and inside the `engine` package.
-//
-// @private
-// @param {module:engine/model/selection~Selection|module:engine/model/documentselection~DocumentSelection} selection
-// The selection based on which the insertion position should be calculated.
-// @param {module:engine/model/model~Model} model Model instance.
-// @param {'auto'|'before'|'after'} [place='auto'] The place where to look for optimal insertion range.
-// The default `auto` value will determine itself the best position for insertion.
-// The `before` value will try to find a position before selection.
-// The `after` value will try to find a position after selection.
-// @returns {module:engine/model/range~Range} The optimal range.
+/**
+ * Returns a model range which is optimal (in terms of UX) for inserting a widget block.
+ *
+ * For instance, if a selection is in the middle of a paragraph, the collapsed range before this paragraph
+ * will be returned so that it is not split. If the selection is at the end of a paragraph,
+ * the collapsed range after this paragraph will be returned.
+ *
+ * Note: If the selection is placed in an empty block, the range in that block will be returned. If that range
+ * is then passed to {@link module:engine/model/model~Model#insertContent}, the block will be fully replaced
+ * by the inserted widget block.
+ *
+ * **Note:** Use {@link module:widget/utils#findOptimalInsertionRange} instead of this function outside engine.
+ * This function is only exposed to be used by {@link module:widget/utils#findOptimalInsertionRange findOptimalInsertionRange()}
+ * in the `widget` package and inside the `engine` package.
+ *
+ * @param selection The selection based on which the insertion position should be calculated.
+ * @param model Model instance.
+ * @param place The place where to look for optimal insertion range.
+ * The default `auto` value will determine itself the best position for insertion.
+ * The `before` value will try to find a position before selection.
+ * The `after` value will try to find a position after selection.
+ * @returns The optimal range.
+ */
 export function findOptimalInsertionRange(
```

Comment:
Should this be marked as internal?

Reply:
IMO, no.

It is re-exported in `ckeditor5-engine/src/index.ts`.